omroneipApp/src/drvOmroneip.h

```diff
@@ -0,0 +1,138 @@
+/* ANSI C includes  */
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <iostream>
+#include <unordered_map>
+#include <memory>
+#include <algorithm>
+#include <vector>
+#include <list>
+#include <array>
+#include <chrono>
+#include <fstream>
+
+/* EPICS includes */
+#include <dbAccess.h>
+#include <epicsStdio.h>
+#include <epicsString.h>
+#include <epicsThread.h>
+#include <epicsMutex.h>
+#include <epicsEvent.h>
+#include <epicsTime.h>
+#include <alarm.h>
+#include <epicsEndian.h>
+#include <epicsExit.h>
+#include <cantProceed.h>
+#include <errlog.h>
+#include <osiSock.h>
+#include <iocsh.h>
+#include <initHooks.h>
+#include <epicsExport.h>
+
+/* libplctag include */
+#include "libplctag.h"
+
+/* Asyn includes */
+#include "asynPortDriver.h"
+#include "asynOctetSyncIO.h"
+#include "asynCommonSyncIO.h"
+#include "asynParamType.h"
+
+#define CREATE_TAG_TIMEOUT 1000 //ms
+
+typedef enum {
+  dataTypeBool,
+  dataTypeInt,
+  dataTypeDInt,
+  dataTypeLInt,
+  dataTypeUInt,
+  dataTypeUDInt,
+  dataTypeULInt,
+  dataTypeUInt_BCD,
+  dataTypeUDInt_BCD,
+  dataTypeULInt_BCD,
+  dataTypeReal,
+  dataTypeLReal,
+  dataTypeString,
+  dataTypeWord,
+  dataTypeDWord,
+  dataTypeLWord,
+  dataTypeUDT,
+  dataTypeTIME,
+  MAX_OMRON_DATA_TYPES
+} omronDataType_t;
+
+struct omronDrvUser_t;
+class omronEIPPoller;
+
+/* Main class for the driver */
+class epicsShareClass drvOmronEIP : public asynPortDriver {
+public:
+  drvOmronEIP(const char *portName,
+              char *gateway,
+              char *path,
+              char *plcType,
+              int debugLevel);
+  ~drvOmronEIP();
+  bool omronExiting_;
+
+  void readPoller();
+  /* Takes a csv style file, where each line contains a structure name followed by a list of datatypes within the struct
```

Comment:
I would suggest moving these comments to the function in the .cpp file and format them as doxygen. That seems to be most common place to put this although Doxygen *might* support docs-comments in both header and source...

Reply:
I will do something like Doxygen at some point, but will create a seperate branch for it